src/main/java/capstone/mfslbackend/service/FantasyLeagueService.java

```diff
@@ -91,4 +98,117 @@ public List<FantasyLeaguePlayer> getFantasyLeaguePlayers(Long leagueId, List<Map
+
+                if (match == 0) {
+                    awayTeam = numTeams - 1;
+                }
+
+                Long homeTeamId = teams.get(homeTeam).getId();
+                Long awayTeamId = teams.get(awayTeam).getId();
+
+
+                // Exclude matches involving the ghost team
+                if (!hasGhostTeam || (homeTeamId != -1L && awayTeamId != -1L)) {
+                    matches.add(homeTeamId);
+                    matches.add(awayTeamId);
+                }
+            }
+            schedule.add(new ArrayList<>(matches));
+
+        }
+        int w =0;
+        // Create the weeks
+        for (int matchIndex = 0; matchIndex < matches.size(); matchIndex += 2) {
+            FantasyWeek fantasyWeek = new FantasyWeek();
+            Long homeTeamId = matches.get(matchIndex);
+            Long awayTeamId = matches.get(matchIndex + 1);
+
+            FantasyTeam teamA = fantasyTeamRepository.findFantasyTeamById(homeTeamId);
+            FantasyTeam teamB = fantasyTeamRepository.findFantasyTeamById(awayTeamId);
+
+            Set<FantasyWeek> weeksA = teamA.getFantasyWeeksA();
+            Set<FantasyWeek> weeksB = teamB.getFantasyWeeksB();
+            fantasyWeek.setFantasyTeamA(teamA);
+            fantasyWeek.setFantasyTeamB(teamB);
+            weeksA.add(fantasyWeek);
+            weeksB.add(fantasyWeek);
+            teamA.setFantasyWeeksA(weeksA);
+            teamB.setFantasyWeeksB(weeksB);
+
+            //fix order number once decided
+            teamA.setOrderNumber(matchIndex);
+            teamB.setOrderNumber(matchIndex + 1);
+
+            teamB.setOpponentNumber(Math.toIntExact(homeTeamId));
+            teamA.setOpponentNumber(Math.toIntExact(awayTeamId));
+            fantasyWeekRepository.save(fantasyWeek);
+        }
+        return getFantasyWeeksByLeagueId(leagueId);
+
+    }
+
+
+
+public List<FantasyWeek> getFantasyLeagueMatchups(Long leagueId, int weekNumber) {
+
+        List<FantasyWeek> schedule = getFantasyWeeksByLeagueId(leagueId);
+        List<FantasyWeek> leagueMatchups = new ArrayList<>();
+
+        if (weekNumber < 1 || weekNumber > schedule.size()) {
+            for (FantasyWeek fantasyW : schedule) {
+                int newWeekNumber = fantasyW.getWeekNumber() % schedule.size();
+                if (newWeekNumber == 0) {
+                    newWeekNumber = schedule.size(); // If modulo is 0, set week number to the maximum
+                }
+                fantasyW.setWeekNumber(newWeekNumber);
+                fantasyWeekRepository.save(fantasyW);
+            }
+        }
+
+        for (FantasyWeek fantasyW : schedule) {
+            if (fantasyW.getWeekNumber() == weekNumber) {
+                leagueMatchups.add(fantasyW);
+            }
+        }
+
+        return leagueMatchups;
+    }
+
+    public List<FantasyWeek> getFantasyWeeksByLeagueId(Long leagueId) {
+        List<FantasyTeam> teams = fantasyTeamRepository.findFantasyTeamsByFantasyLeagueId(leagueId);
```

Comment:
dont call the repository directly. theres a service function that throws a 404 if theres no object so that way we don't have a Null pointer exception

Reply:
you can just get the league from the getFantasyLeagueById service funtion and do fantasyLeague.getFantasyTeams()